components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/EndpointResource.java

```diff
@@ -142,8 +169,48 @@ private JSONObject getEndpointAsJson(Endpoint endpoint) {
         JSONObject endpointObject = endpoint.getJsonRepresentation();
         OMElement synapseConfiguration = EndpointSerializer.getElementFromEndpoint(endpoint);
         endpointObject.put(Constants.SYNAPSE_CONFIGURATION, synapseConfiguration);
+        endpointObject.put(IS_ACTIVE, isEndpointActive(endpoint));
 
         return endpointObject;
     }
 
+    /**
+     * Changes the endpoint state based on json payload.
+     *
+     * @param axis2MessageContext Axis2Message context
+     * @param configuration       Synapse configuration
+     * @param payload             Request json payload
+     */
+    private void changeEndpointStatus(org.apache.axis2.context.MessageContext axis2MessageContext,
+                                      SynapseConfiguration configuration, JsonObject payload) {
+
+        String endpointName = payload.get(NAME).getAsString();
+        String status = payload.get(STATUS).getAsString();
+        Endpoint ep = configuration.getEndpoint(endpointName);
+        if (ep != null) {
+            JSONObject jsonResponse = new JSONObject();
+            if (INACTIVE_STATUS.equalsIgnoreCase(status)) {
+                ep.getContext().switchOff();
+                jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, endpointName + " : is switched off");
+            } else if (ACTIVE_STATUS.equalsIgnoreCase(status)) {
+                ep.getContext().switchOn();
+                jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, endpointName + " : is switched ON");
+            } else {
+                LOG.warn("Specified Status: " + status + " is not valid");
+                jsonResponse.put(Constants.MESSAGE_JSON_ATTRIBUTE, "Provided state is not valid");
+            }
+            Utils.setJsonPayLoad(axis2MessageContext, jsonResponse);
+        } else {
+            LOG.warn("Specified Endpoint: " + endpointName + " does not exist");
+            Utils.setJsonPayLoad(axis2MessageContext, Utils.createJsonErrorObject("Endpoint does not exist"));
+        }
+    }
+
+    private Boolean isEndpointActive(Endpoint endpoint) {
+        // 1 represents the endpoint active state
+        if (endpoint.getContext().isState(1)) {
+            return true;
+        }
+        return false;
```

Comment:
Shall we simplify as return endpoint.getContext().isState(1);

Reply:
done